### ISSUE TITLE
[469a][guix] No available video device / segfault

### ISSUE THREAD

**Issue by Ambrevar** [OldUnreal/UnrealTournamentPatches#55]
I'm trying to package Unreal Tournament for [Guix](http://guix.gnu.org/).
I tried two strategies: either reuse the embedded libraries or use the Guix ones
(SDL2, SDL2-ttf, openal, libxmp).

By the way, libfreetype is not included with the game, so I had to include
Guix' `freetype` as a dependency in all cases.

I can't start the game for now.

- Using UT embedded SDL library:

```
WARNING: using preference directory: /home/ambrevar/.utpg/System/
WARNING: Using system ini file: /home/ambrevar/.utpg/System/UnrealTournament.ini
WARNING: Using user ini file: /home/ambrevar/.utpg/System/User.ini
Couldn't initialize SDL: No available video device


History: main

Exiting due to error
```

- It I use Guix' SDL library, I get the splash screen, but then it fails with:

```
WARNING: using preference directory: /home/ambrevar/.utpg/System/
WARNING: Using system ini file: /home/ambrevar/.utpg/System/UnrealTournament.ini
WARNING: Using user ini file: /home/ambrevar/.utpg/System/User.ini

Developer Backtrace:
[ 1]  /UnrealTournament/System/./Core.so(_Z12HandleSignali+0x208) [0xf7a29d08]
[ 2]  linux-gate.so.1(__kernel_sigreturn+0) [0xf7fc3580]
[ 3]  /UnrealTournament/System/./SDLDrv.so(_ZN12USDLViewport14ResizeViewportEjiii+0x19d) [0xf01a142d]
[ 4]  /UnrealTournament/System/./SDLDrv.so(_ZN12USDLViewport11UpdateInputEi+0x7da) [0xf01a085a]
[ 5]  /UnrealTournament/System/./Engine.so(_ZN6UInput10ResetInputEv+0x297) [0xf7d18de7]
[ 6]  /UnrealTournament/System/./Engine.so(_ZN6UInput4InitEP9UViewport+0x52) [0xf7d17a62]
[ 7]  /UnrealTournament/System/./Engine.so(_ZN11UGameEngine4InitEv+0x711) [0xf7d062a1]
[ 8]  ./ut-bin-x86(main+0xb06) [0x804e656]
[ 9]  /gnu/store/9728x9zhj8d3zlyzsjpz342jpccrgkgd-glibc-2.31/lib/libc.so.6(__libc_start_main+0x106) [0xf731a936]
[10]  ./ut-bin-x86() [0x804d8bf]
Signal: SIGSEGV [segmentation fault]
Aborting.

Signal: SIGSEGV [segmentation fault]

History: USDLViewport::ResizeViewport <- USDLViewport::UpdateInput <- UInput::ResetInput <- UInput::Init <- UGameEngine::Init <- InitEngine <- main

Exiting due to error
```

Any clue what could be happening?

Note that Guix is a non-FHS distribution.  This may explain why in the first
case the embedded SDL fails to find the video device because it's looking for
drivers in places that don't exist with Guix.

**Comment by Ambrevar** [comment]
Guix' SDL2 version is 2.0.12.

**Comment by stijn-volckaert** [comment]
Is this on the latest (and fully up to date) guix for x86_64?

**Comment by Ambrevar** [comment]
Yes! :)

**Comment by stijn-volckaert** [comment]
![afbeelding](https://user-images.githubusercontent.com/10390414/197339475-124ddc51-6c48-456d-b939-8460a616bc72.png)

I tried running the upcoming UT v469c 64-bit patch on Guix 1.3.0 today and it seems to work. There are some caveats, though:
- I had to manually patch the interpreter of the main binaries to point them to the correct ld-linux.so.2
- I had to delete the SDL2, openal, xmp, and mpg123 libraries included in the patch and had to install guix' packages instead. I also had to set up symlinks in the UnrealTournament/System64/ folder to point to the guix packages
- I couldn't change my game resolution from the in-game menu. I'm pretty sure this is because I'm running guix in a VMware VM without the VMware tools installed
- Some textures that are meant to be fully opaque were rendered semi-transparently. I have no idea why. Could be a driver/VMware issue
